mantidproject/mantid
# ReplaceSpecialValues: extended checks to the error axis

**Description of work.**



Performed a minimum intervention in the ReplaceSpecialValues algorithm to not only check whether Y-axis contains some unwanted values but also to check the Error-axis.  

**To test:**

Test that the algorithm picks up that the Error axis contains unwanted values, e.g. NaN and whether it replaces it with the desired parameter:

For example:
1. Open the Mantid workbench 
2.  Create a workspace with inf, nan, or other value that one desires no to be there in the in the error axis data, while the Y-axis should contain valid numbers.
3.  Run the ReplaceSpecialValues algorithm on this workspace.
4.  Observe whether the chosen type of issue (inf/nan/other) is fixed.



Fixes #29181 

Changes added to the release note and algorithm documentation.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
